libraries/AP_Scripting/examples/quadruped.lua

```diff
@@ -0,0 +1,295 @@
+    local leg_step = gait_step - gait_step_leg_start[moving_leg]
+    local move_requested = (math.abs(x_speed) > speed_dz) or (math.abs(y_speed) > speed_dz) or (math.abs(yaw_speed) > speed_dz)
+
+    if ((move_requested and (gait_lifted_steps > 0) and leg_step==0) or
+        (not move_requested and leg_step==0 and ((gait_pos_x[moving_leg]>2) or
+        (gait_pos_z[moving_leg]>2) or (gait_rot_z[moving_leg] >2)))) then
+        gait_pos_x[moving_leg] = 0
+        gait_pos_z[moving_leg] = -leg_lift_height
+        gait_pos_y[moving_leg] = 0
+        gait_rot_z[moving_leg] = 0
+
+    elseif (((gait_lifted_steps==2 and leg_step==0) or (gait_lifted_steps>=3 and
+            (leg_step==-1 or leg_step==(gait_step_total-1)))) and move_requested) then
+        gait_pos_x[moving_leg] = -x_speed/gait_lift_divisor
+        gait_pos_z[moving_leg] = -3*leg_lift_height/(3+gait_half_lift_height)
+        gait_pos_y[moving_leg] = -y_speed/gait_lift_divisor
+        gait_rot_z[moving_leg] = -yaw_speed/gait_lift_divisor
+
+    elseif ((gait_lifted_steps>=2) and (leg_step==1 or leg_step==-(gait_step_total-1)) and move_requested) then
+        gait_pos_x[moving_leg] = x_speed/gait_lift_divisor
+        gait_pos_z[moving_leg] = -3*leg_lift_height/(3+gait_half_lift_height)
+        gait_pos_y[moving_leg] = y_speed/gait_lift_divisor
+        gait_rot_z[moving_leg] = yaw_speed/gait_lift_divisor
+
+    elseif (((gait_lifted_steps==5 and (leg_step==-2 ))) and move_requested) then
+        gait_pos_x[moving_leg] = -x_speed/2
+        gait_pos_z[moving_leg] = -leg_lift_height/2
+        gait_pos_y[moving_leg] = -y_speed/2
+        gait_rot_z[moving_leg] = -yaw_speed/2
+
+    elseif ((gait_lifted_steps==5) and (leg_step==2 or leg_step==-(gait_step_total-2)) and move_requested) then
+        gait_pos_x[moving_leg] = x_speed/2
+        gait_pos_z[moving_leg] = -leg_lift_height/2
+        gait_pos_y[moving_leg] = y_speed/2
+        gait_rot_z[moving_leg] = yaw_speed/2
+
+    elseif ((leg_step==gait_down_steps or leg_step==-(gait_step_total-gait_down_steps)) and gait_pos_y[moving_leg]<0) then
+        gait_pos_x[moving_leg] = x_speed/2
+        gait_pos_z[moving_leg] = y_speed/2
+        gait_pos_y[moving_leg] = yaw_speed/2
+        gait_rot_z[moving_leg] = 0
+
+    else
+        gait_pos_x[moving_leg] = gait_pos_x[moving_leg] - (x_speed/gait_speed_divisor)
+        gait_pos_z[moving_leg] = 0
+        gait_pos_y[moving_leg] = gait_pos_z[moving_leg] - (y_speed/gait_speed_divisor)
+        gait_rot_z[moving_leg] = gait_rot_z[moving_leg] - (yaw_speed/gait_speed_divisor)
+    end
+end
+
+-- Body Forward Kinematics calculates where each leg should be.
+-- inputs are
+--   a) body rotations: body_rot_x, body_rot_y, body_rot_z
+--   b) body position: body_pos_x, body_pos_y, body_pos_z
+--   c) offset of the center of body
+function body_forward_kinematics(X, Y, Z, Xdist, Ydist, Zrot)
+    local totaldist = {X + Xdist + body_pos_x, Y + Ydist + body_pos_y}
```

Comment:
Given that you always index the table here with 1 and 2 that doesn't actually seem to be helping from a readability perspective. Given that I'd probably make this a pair of locals, and use that in the rest of the function. It should generate less garbage, and run faster.
```suggestion
    local totaldist_x = X + Xdist + body_pos_x
    local totaldist_y =  Y + Ydist + body_pos_y
```

Reply:
This suggested change has been included in the last commit, thanks!